src/pages/patientView/trialMatch/TrialMatchTableUtils.ts

```diff
@@ -78,16 +88,42 @@ export function groupTrialMatchesByAgeNumerical(armGroup: ITrialMatch[]): IClini
 
 export function groupTrialMatchesByGenomicAlteration(ageGroup: ITrialMatch[]) {
     const matchesGroupedByGenomicAlteration = _.groupBy(ageGroup, (trial: ITrialMatch) => trial.genomicAlteration);
-    const matches: IGenomicGroupMatch[] = []; // positive matches
-    const notMatches: IGenomicGroupMatch[] = []; // negative matches
+    const matches: IGenomicMatchType = {
+        mutationMatches: [],
+        msiMatches: [],
+        cnaMatches: [],
+        wildTypeMatches: []
+    }; // positive matches
+    const notMatches: IGenomicMatchType = {
+        mutationMatches: [],
+        msiMatches: [],
+        cnaMatches: [],
+        wildTypeMatches: []
+    }; // negative matches
     _.forEach(matchesGroupedByGenomicAlteration, (genomicAlterationGroup, genomicAlteration) => {
         const genomicGroupMatch = formatTrialMatchesByMatchType(genomicAlterationGroup, genomicAlteration);
         // If a genomic alteration(i.e. BRAF V600E) contains a "!",
         // it means this trial cannot be used for the genomic alteration, which is a "NOT" match.
         if(genomicAlteration.includes('!')) {
-            notMatches.push(genomicGroupMatch);
+            if (genomicAlterationGroup[ 0 ][ 'matchType' ] === 'MUTATION') {
```

Comment:
these long if/else constructs should be re-written using switch statements

Reply:
or even better, make a table with match type to the corresponding array and then you can just look it up and push into it.

Reply:
Sorry, I'm confused. Do you mean to create tables for different type of trial matches? If yes, that's not what curators want. They want to see trial matches grouped by trial id instead of match type.